python-avd/pyavd/_eos_designs/structured_config/core_interfaces_and_l3_edge/router_bgp.py

```diff
@@ -36,40 +34,32 @@ def router_bgp(self: AvdStructuredConfigCoreInterfacesAndL3EdgeProtocol) -> dict
                 msg = f"{self.data_model}.p2p_links.[].as or {self.data_model}.p2p_links_profiles.[].as"
                 raise AristaAvdInvalidInputsError(msg)
 
-            neighbor = {
-                "remote_as": p2p_link_data["peer_bgp_as"],
-                "peer": p2p_link_data["peer"],
-                "description": p2p_link_data["peer"],
-                "peer_group": self.inputs.bgp_peer_groups.ipv4_underlay_peers.name,
-            }
+            neighbor = EosCliConfigGen.RouterBgp.NeighborsItem(
+                remote_as=p2p_link_data["peer_bgp_as"],
+                peer=p2p_link_data["peer"],
+                description=p2p_link_data["peer"],
+                peer_group=self.inputs.bgp_peer_groups.ipv4_underlay_peers.name,
+            )
 
             # RFC5549
             if self.inputs.underlay_rfc5549 and p2p_link.routing_protocol != "ebgp":
-                neighbor_interfaces.append({"name": p2p_link_data["interface"], **neighbor})
+                self.structured_config.router_bgp.neighbor_interfaces.append_new(
+                    name=p2p_link_data["interface"],
+                    remote_as=p2p_link_data["peer_bgp_as"],
+                    peer=p2p_link_data["peer"],
+                    description=p2p_link_data["peer"],
+                    peer_group=self.inputs.bgp_peer_groups.ipv4_underlay_peers.name,
+                )
                 continue
 
             # Regular BGP Neighbors
             if p2p_link_data["ip"] is None or p2p_link_data["peer_ip"] is None:
                 msg = f"{self.data_model}.p2p_links.[].ip, .subnet or .ip_pool"
                 raise AristaAvdInvalidInputsError(msg)
 
-            neighbor["bfd"] = p2p_link.bfd
+            neighbor.bfd = p2p_link.bfd
             if p2p_link_data["bgp_as"] != self.shared_utils.bgp_as:
-                neighbor["local_as"] = p2p_link_data["bgp_as"]
-
-            # Remove None values
-            neighbor = {key: value for key, value in neighbor.items() if value is not None}
-
-            neighbors.append({"ip_address": get_ip_from_ip_prefix(p2p_link_data["peer_ip"]), **neighbor})
-
-        router_bgp = {}
-        if neighbors:
-            router_bgp["neighbors"] = neighbors
-
-        if neighbor_interfaces:
-            router_bgp["neighbor_interfaces"] = neighbor_interfaces
-
-        if router_bgp:
-            return router_bgp
+                neighbor.local_as = p2p_link_data["bgp_as"]
```

Comment:
this can be put in the neighbor creaton by doing

```
local_as= p2p_link_data["bgp_as"] if p2p_link_data["bgp_as"] != self.shared_utils.bgp_as else None
```

Reply:
Fixed!